modules/serverless/pages/serving/serving-transport-encryption.adoc

```diff
@@ -0,0 +1,446 @@
+* Specific SANs are used to verify each connection. Each component needs to trust the CA that signed the certificates. For this, {ServerlessProductName} system components will consume and trust a bundle of CAs. The CA bundle needs to be provided by the cluster administrator.
+
+
+== Identify and select an appropriate certificate issuer [[issuer_selection]]
+
+[INFO]
+====
+Issuers here are referring to cert-manager `Issuers` and `ClusterIssuers`. They represent certificate authorities (CAs) that can generate signed certificates by honoring certificate signing requests.
+Reference: https://cert-manager.io/docs/concepts/issuer/
+====
+
+In general, you can refer to the cert-manager documentation (https://cert-manager.io/docs/configuration/issuers/). There are examples available for:
+
+* A custom CA stored in a K8s secret
+* HTTP-01 challenges, e.g. Let's encrypt
+* DNS-01 challenges
+* Self-signed issuers
+
+[IMPORTANT]
+====
+Please note, that not all issuer types work for each Knative Serving encryption feature. See below for details.
+====
+
+Depending on the feature you would like to use, {serverlessproductname} requires your certificate issuer to be capable of signing the following certificates:
+
+For the `cluster-local-domain-tls` feature, the issuer needs to be able to sign certificates for cluster-local domains like:
+
+* `myapp.<namespace>`
+* `myapp.<namespace>.svc`
+* `myapp.<namespace>.svc.cluster.local`
+
+As the CA usually is not within the cluster, verification via ACME protocol (DNS01/HTTP01) is not possible. You can use an issuer that allows creating these certificates (e.g. cert-manager's CA issuer).
+
+For the `system-internal-tls` feature, the CA must be able to sign certificates with a subject alternative name (SAN) of:
+
+* `kn-routing`
+* `kn-user-<namespace>` (<namespace> is each namespace where Knative Services are/will be created)
+* `data-plane.knative.dev`
+
+These SANs are needed for Knative to verify connections between the internal components.
+As this is also not possible via ACME protocol (DNS01/HTTP01), you need to configure an issuer that allows creating these certificates (e.g. cert-manager's CA issuer).
+
+
+== Installing and configuring {ServerlessProductName} transport encryption
+
+=== Prerequisites
+
+* You have access to an {product-title} account with cluster administrator access.
+
+* Install the OpenShift CLI (`oc`).
+
+* Install the {certmanageroperatorname}.
+
+* Install the {serverlessoperatorname}.
+
+[IMPORTANT]
+====
+If you have installed the {serverlessoperatorname} before installing the {certmanageroperatorname}, you will have to restart the following components to enable the Knative Serving cert-manager integration. If this is not done, Knative will not create the necessary cert-manager resources, leading to pending Knative Services.
+
+* Controller deployment in the `knative-serving` namespace.
+* Activator deployment in the `knative-serving` namespace.
+====
```

Comment:
q: why isn't this driven by the configs in "Understanding and configuring the transport encryption configuration" ?

It might be against or not working well in GitOps only environments, maybe the operator could restart those deployments automatically based on the `config-certmanager` CM value

Reply:
We intend to resolve these limitations in a future release. Its more a question on where we invest that time. Either making the restart unnecessary or doing this in the operator (which currently does not watch or know about that state). IMHO this should be doable for the time being dev-preview even with GitOps, as it is just a one-time thing.

Reply:
Does argo support that? https://github.com/argoproj/argo-cd/issues/13642